### ISSUE TITLE
Console does not respect codeCellConfig

### ISSUE THREAD

**Issue by CGamesPlay** [jupyterlab/jupyterlab#6352]
**Describe the bug**
The lab feature "Console" (unsaved notebook) does not respect the configuration for code cells in notebooks.

**To Reproduce**

1. Using Jupyter Lab advanced settings editor, set `codeCellConfig.autoClosingBrackets` to `true`.
2. Open a new Notebook and type `[` into the first cell.
3. See that `[]` is written into the cell.
3. Open a new Console and type `[` into the cell.
4. See that `[]` is written in the cell.
5. Change `codeCellConfig.autoClosingBrackets` to `false`.
6. Open a new Notebook and type `[` into the first cell.
7. See that `[` is written into the cell.
8. Open a new Console and type `[` into the cell.
9. See that `[]` is written in the cell.

**Desktop (please complete the following information):**

- Browser Safari Version 12.1 (14607.1.40.1.4)
- JupyterLab Version 0.35.6

**Additional context**

<details><summary>Troubleshoot Output</summary>
https://gist.github.com/CGamesPlay/0d0b6eb698a34b2bc5dbb8c0e17251b0
</details>

**Comment by mlucool** [comment]
Today there is an option:
```
    // Interaction mode
    // Whether the console interaction mimics the notebook
    // or terminal keyboard shortcuts.
    "interactionMode": "notebook",
```
This makes it feel like it's ok just to honor all settings in notebook.

**Comment by goanpeca** [comment]
@jasongrout, do you agree with the suggestion from above?

That should already handle that.

**Comment by jess-x** [comment]
Hi @jasongrout, I am currently working on this one. Is it ok to mirror notebook's autoClosingBrackets setting when Console's `interactionMode` is in "notebook"?

**Comment by jasongrout** [comment]
I would suggest mirroring it always, not just when the interaction mode is "notebook"

**Comment by jasongrout** [comment]
The interaction mode being notebook just means (right now) that it executes on shift-enter rather than enter. I suppose we could extend that, but I think the direction I'd love to see is mirroring codemirror settings across all codemirrors.

**Comment by jess-x** [comment]
> I think the direction I'd love to see is mirroring codemirror settings across all codemirrors.

I agree, having an unified codemirror setting for all codemirrors seems intuitive to me (even though I personally mainly interact with codemirrors through notebooks) 
I noticed only notebooks have an interface to set codeCellConfig, right now. Does this translate to making a user change in notebook's codeCellConfig should broadcast to all codemirrors (including ones in Console)?

**Comment by jasongrout** [comment]
I'm hoping that with 4.0's planned move to codemirror 6, we can unify codemirror settings across the places codemirror is used in the codebase.

I would not tie the codemirror config to the 'interaction mode: notebook' setting - that was designed for changing how you could run code in the console, not how the code editor behaved. (I can see maybe we chose a name that was too generic?). In other words, I can see people wanting to have the code editor setting independent of whether they use enter or shift-enter to run code.

It might be easiest for us to wait until 4.0, and work on a set of unified settings for codemirrors across the application. If we need the setting before 4.0, and if there is an elegant way for the console to mirror the notebook code editor settings, I think that would be great. I'm not sure there is really a clean way to do that, though. Barring that, implementing a separate option for console code editor settings might be the easiest way to move things forward now.

**Comment by JasonWeill** [comment]
@jasongrout Is this still an issue, considering that the move to CodeMirror 6 has been committed since your most recent comment?

**Comment by jasongrout** [comment]
When I test 4.0a32, I see the solution @jess-x implemented in https://github.com/jupyterlab/jupyterlab/pull/10555, namely that consoles now have a separate configuration for their prompt cell that mirrors the options available for notebook code cells. I don't see a unified single place where codemirror settings are set and apply to both code cells in the notebook and the prompt cell in a console, which is the idea that was discussed above.

I think this solution is great for jlab 4, but I think it also merits consideration in the future of having a single config for the code editor that serves as a default for both notebook code cells and the console prompt cell. In other words, if someone selects "auto-closing brackets", I think they typically want that setting applied in both places, and right now they have to check that setting in two different places.

I'll move it off the 4.0 milestone to Future, though.
